site/en/docs/android/custom-tabs/guide-warmup-prefetch/index.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Warm-up and pre-fetching: using the Custom Tabs Service"
+seoTitle: "Guide: warm-up and pre-fetching - using the Custom Tabs Service"
+date: 2023-04-21
+description: Learn how to make sure that your Custom Tabs start as quickly as possible.
+authors:
+  - sebastianbenz
+---
+
+The third part of this guide focuses on warming up the browser process via [warmup()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#warmup(long)) and prefetching web pages via [mayLaunchUrl](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession#mayLaunchUrl(android.net.Uri,android.os.Bundle,java.util.List%3Candroid.os.Bundle%3E))(). Warming up the browser process can save up to 700ms when opening a link. Pre-rendering content via mayLaunchUrl will make external content open instantly. Together both APIs can greatly improve the user experience of a Custom Tabs integration and are highly recommended.
+
+The basic steps are:
+
+1. Check via [CustomTabsClient](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient).[getPackageName()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#getPackageName(android.content.Context,java.util.List%3Cjava.lang.String%3E)) If the default browser supports Custom Tabs. If yes, bind to the CustomTabsService via [CustomTabsClient.bindCustomTabsService()](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#bindCustomTabsService(android.content.Context,java.lang.String,androidx.browser.customtabs.CustomTabsServiceConnection)).
```

Comment:
```suggestion
1. Check via [`CustomTabsClient`](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient).[`getPackageName()`](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#getPackageName(android.content.Context,java.util.List%3Cjava.lang.String%3E)) If the default browser supports Custom Tabs. If yes, bind to the CustomTabsService via [`CustomTabsClient.bindCustomTabsService()`](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsClient#bindCustomTabsService(android.content.Context,java.lang.String,androidx.browser.customtabs.CustomTabsServiceConnection)).
```

Reply:
>If the default browser supports Custom Tabs.

This isn't a complete sentence. Is this supposed to be part of the previous sentence or is something else missing?

Reply:
part of the previous sentence, fixed